cmd/headers/download/downloader.go

```diff
@@ -216,6 +216,7 @@ func NewStagedSync(
 			nil,
 			nil,
 			controlServer.chainConfig,
+			pruningDistance,
```

Comment:
Don't think you can use `pruningDistance` here, in fact I don't think it should be the variable on this level. It is currently controlled by `--storage-mode h` and you don't want to be dependent on that

Reply:
On the one hand I had the same thoughts, but then I realised that transaction status is a kinda history and I'd like to have 90000 blocks by default.
Can do a separate variable or flag.

Reply:
No, it is not the same as history. We do want to prune the intermediate TEVM status regardless of the settings. Always. That is how I did pruning of intermediate tables for trace call set